extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -89,6 +91,6 @@ public void build(CombinedIndexBuildItem indexBuildItem, BuildProducer<Reflectiv
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, RangeAssignor.class.getName()));
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, RoundRobinAssignor.class.getName()));
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, StickyAssignor.class.getName()));
-
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, BeforeDestroyed.class.getName()));
```

Comment:
Is it maybe something worth registering globally? I imagine other libraries may try to do something before a specific scope is destroyed too...

Reply:
I think it makes sense to move this to the Arc module

Reply:
moved. Thanks